speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DataCompressionMenu.cpp

```diff
@@ -0,0 +1,233 @@
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Write data compression type
+    char val[32];
+    sprintf(val, "%d", m_dataCompressionType);
+    GfParmSetStr(readParam, PRM_DATA_COMPRESSION, GFMNU_ATTR_SELECTED, val);
+
+    // Save custom compression to xml file
+    char buf[32];
+    sprintf(buf, "%d", m_customCompressionLevel);
+    GfParmSetStr(readParam, PRM_CUSTOM_COMPRESSION, GFMNU_ATTR_TEXT, buf);
+
+    // Write queued changes
+    GfParmWriteFile(nullptr, readParam, COMPRESSION_SCREEN_NAME);
+}
+
+/// @brief Saves the settings so the mediator (or future instances) can access them
+static void SaveSettings()
+{
+    int compressionLevel;
+    switch (m_dataCompressionType)
+    {
+        case COMPRESSION_NONE:
+        {
+            compressionLevel = 1;
+            break;
+        }
+        case COMPRESSION_MINIMUM:
+        {
+            compressionLevel = 3;
+            break;
+        }
+        case COMPRESSION_MEDIUM:
+        {
+            compressionLevel = 5;
+            break;
+        }
+        case COMPRESSION_MAXIMUM:
+        {
+            compressionLevel = 9;
+            break;
+        }
+        default:  // Custom
+        {
+            compressionLevel = m_customCompressionLevel;
+            break;
+        }
+    }
+
+    SMediator* mediator = SMediator::GetInstance();
+    mediator->SetCompressionLevel(compressionLevel);
+
+    SaveSettingsToFile();
+}
+
+/// @brief Takes actions that need to be run on activation of the data selection screen
+static void OnActivate(void* /* dummy */)
+{
+    LoadSettings();
+}
+
+/// @brief Switches back to the data selection menu
+static void SwitchToDataSelectionMenu(void* /* dummy */)
+{
+    // go back to the researcher screen
+    GfuiScreenActivate(s_prevHandle);
+}
+
+/// @brief Saves the settings and then switched back to the data selection menu
+static void SaveAndGoBack(void* /* dummy */)
+{
+    SaveSettings();
+
+    SwitchToDataSelectionMenu(nullptr);
+}
+
+/// @brief        Sets the type of data compression
+/// @param p_info Information about the radio button list
+static void SelectCompression(tRadioButtonInfo* p_info)
+{
+    m_dataCompressionType = static_cast<DataCompressionType>(p_info->Selected);
+}
+
+/// @brief Handle input in the custom compression type textbox
+static void SetCompressionLevel(void*)
+{
+    char* val = GfuiEditboxGetString(s_scrHandle, m_customCompressionLevelControl);
+    char* endptr;
+    m_customCompressionLevel = (int)strtol(val, &endptr, 0);
+    if (*endptr != '\0')
+        std::cerr << "Could not convert " << val << " to long and leftover string is: " << endptr << std::endl;
+
+    // the compression level needs to be uneven
+    if (m_customCompressionLevel % 2 == 0)
+    {
+        m_customCompressionLevel++;
+    }
+
+    Clamp(m_customCompressionLevel, 1, MAX_COMPRESSION_LEVEL);
```

Comment:
Why is this clamped to a MAX_COMPRESSION_LEVEL of 49? The tooltip of the custom setting states that this is a percentage, so why is this clampedto 49%?

The radiobuttons also state that NONE = 0% compression, which is somehow compressionlevel 1, but MINIMUM = 66% compression, which is a level 3. Following this pattern I don't think 100% compression ends up at a level of 49 (but correct me if i'm wrong, because I don't know what exactly this 'level' is supposed to mean) 

It might even be more clear to store the compression percentage instead of some meaningless 'level'? Otherwise explain exactly what this level is.

Reply:
I will change the tooltip, because compressionLevel is the number of timesteps from which you take the average when storing the data. So the number of timesteps you don't save data.